url.ts

```diff
@@ -0,0 +1,36 @@
+const stripSlash = (path: string) => {
+  const leadingSlash = path.charCodeAt(0) === 47
+  const trailingSlash = path.charCodeAt(path.length - 1) === 47
+  const start = leadingSlash ? 1 : 0
+  path = path.slice(start)
+  if (path.length && trailingSlash) path = path.slice(0, path.length - 1)
+  return path
+}
+
+/**
+ * This function joins a list of paths with / (forward slash)
+ * @param {boolean} padLeft - pads the final path with leading slash if it does not exist
+ * @param {boolean} padRight - pads the final path with trailling slash if it does not exist
+ * @param {string[]} [path] - array of paths.
+ */
+const pathJoin = (padLeft: boolean, padRight: boolean, ...path: string[]) => {
```

Comment:
why not use `new URL` instead? or std/path?

Reply:
`new URL` expects a standard URL
```ts
> new URL('/route')
Uncaught TypeError: Invalid URL: '/route'
    at getSerialization (ext:deno_url/00_url.js:88:11)
    at new URL (ext:deno_url/00_url.js:383:27)
    at <anonymous>:2:1
```

`path.join` joins with a platform specific delimiter. In linux and macos, this is `/` but in windows it is `\`. See https://stackoverflow.com/a/45575007

Reply:
@aarontravass you can use `std/path/posix`

Reply:
tbh having a custom function is ok since the one in std is meant for fs